new types + adddressing some comments (#1)
montycheese/base-web: apps/base-docs/docs/pages/identity/smart-wallet/guides/linked-accounts/creating-sub-accounts.mdx

# Creating Sub Accounts

After setting up your project with [Coinbase Smart Wallet integration](/identity/smart-wallet/guides/linked-accounts/setup), the next step is implementing sub account creation. Sub accounts allow your application to create wallet accounts that are linked to the user's main Coinbase Smart Wallet.

## What you'll achieve

By the end of this guide, you will:

- Configure the Coinbase Smart Wallet SDK to support sub account creation
- Implement state management for tracking sub accounts in your application

> - Create a user interface that allows users to generate linked sub accounts on demand

Reviewer: ```suggestion
- Create a user interface that allows users to create a sub account on demand
```

## Signer Choice

> Currently browser keys, turnkey are supported for signer for sub account.

Reviewer: ```suggestion
Currently p256 keys, ethereum addresses, and webauthn keys are supported as signer for Sub Accounts.
```

The Coinbase Smart Wallet SDK provides a browser key signer out of the box, which handles the creation and caching of signers for you. In this guide, we'll use the browser key signer to create a sub account.

## Configuring the SDK for Sub Accounts

> Include sub account configuration in your SDK initialization. In this react web app example, we are passing in `getCryptoKeyAccount` from the SDK for the `subaccount.getSigner` param when initializing the SDK in an react effect hook.

Reviewer: ```suggestion
Include Sub Account configuration in your SDK initialization. In this react web app example, we are passing in `getCryptoKeyAccount` from the SDK for the `subaccount.getSigner` param when initializing the SDK in an React useEffect hook.
```

## Adding Sub Account State Management

> Next, extend your context to manage the state and functions needed to create and manage sub accounts.

Reviewer: ```suggestion
Next, extend your context to manage the state and functions needed to create and manage Sub Accounts.
```

## Updating the UI

> Now let's update your main page to include sub account creation, using the context you created in the previous step.

Reviewer: ```suggestion
Now let's update your main page to include Sub Account creation, using the context you created in the previous step.
```
